src/Plugin/inmail/Handler/MailhandlerComment.php

```diff
@@ -0,0 +1,191 @@
+      '#type' => 'item',
+      '#markup' => $this->t('Post comments via email.'),
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function invoke(MessageInterface $message, ProcessorResultInterface $processor_result) {
+    try {
+      $result = $this->getMailhandlerResult($processor_result);
+
+      if ($result->getEntityType() != 'comment') {
+        // Do not run this handler in case we are not dealing with comments.
+        return;
+      }
+
+      // Create a comment.
+      $comment = $this->createComment($message, $result);
+
+      $processor_result->log('CommentHandler', '@comment has been created by @user.', ['@comment' => $comment->label(), '@user' => $comment->getAuthorName()]);
+    }
+    catch (\Exception $e) {
+      // Log error in case verification, authentication or authorization fails.
+      $processor_result->log('CommentHandler', $e->getMessage());
+    }
+  }
+
+  /**
+   * Returns a Mailhandler analyzer result instance.
+   *
+   * @param \Drupal\inmail\ProcessorResultInterface $processor_result
+   *   The result and log container for the message, containing the message
+   *   deliverer and possibly analyzer results.
+   * @return \Drupal\mailhandler_d8\MailhandlerAnalyzerResultInterface
+   *   The Mailhandler analyzer result instance.
+   *
+   * @throws \Exception
+   *   Throws an exception in case there is no Mailhandler analyzer result
+   *   object created. It happens in case all Mailhandler analyzers are
+   *   disabled.
+   */
+  public function getMailhandlerResult(ProcessorResultInterface $processor_result) {
+    /** @var \Drupal\mailhandler_d8\MailhandlerAnalyzerResultInterface $result */
+    $result = $processor_result->getAnalyzerResult(MailhandlerAnalyzerResult::TOPIC);
+
+    // @todo: Remove when support for core Inmail result objects is implemented.
+    if (!$result) {
+      throw new \Exception('Mailhandler Analyzer result object cannot be ensured.');
+    }
+
+    return $result;
+  }
+
+  /**
+   * Creates a new comment from given mail message.
+   *
+   * @param \Drupal\inmail\MIME\MessageInterface $message
+   *   The mail message.
+   * @param \Drupal\mailhandler_d8\MailhandlerAnalyzerResultInterface $result
+   *   The analyzer result.
+   *
+   * @return \Drupal\comment\Entity\Comment
+   *   The created comment.
+   *
+   * @throws \Exception
+   *   Throws an exception in case user is not authorized to create a comment.
+   */
+  protected function createComment(MessageInterface $message, MailhandlerAnalyzerResultInterface $result) {
+    $subject = $result->getSubject();
+    if (!preg_match('/^\[#(\d+)\]\s+/', $subject, $matches)) {
+      throw new \Exception('Referenced entity ID of the comment could not be identified.');
+    }
+
+    // Get a node ID and update the subject.
+    $node_id = $matches[1];
+    $subject = str_replace(reset($matches), '', $subject);
+
+    // Validate whether user is allowed to post comments.
+    $user = $this->validateUser($result);
```

Comment:
Seems like ```validate()``` doesn't "validate" entity ID, so I added custom validation for entity ID and bundle.